test/scale/_size.scss

```diff
@@ -0,0 +1,159 @@
+@use '../../sass/scale/units';
+@use '../../node_modules/sass-true' as *;
+@use '../../sass/scale/size';
+@use '../../sass/scale/config';
+@use '../../sass/tokens';
+@use 'sass:math';
+@use './test-data' as data;
+
+$_global: config.$sizes;
+config.$sizes: data.$test-sizes;
+
+// Scale Tests
+// ===========
+@include test-module('Scale: Size') {
+  // Size [function]
+  // ---------------
+  @include describe('size [function]') {
+    @include it('Get size from map setting') {
+      @include assert-equal(size.size('root'), 20px);
+    }
+
+    @include it('Get size from unit adjustment') {
+      @include assert-equal(size.size('text'), 1rem);
+    }
+
+
+     @include it('Get size from linear adjustment') {
+      @include assert-equal(size.size('h3'), 2rem);
+    } 
+
+    @include it('Get size from ratio adjustment') {
+      @include assert-equal(size.size('big'), 7.59375rem);
+    }
+
+    @include it('Get size from negative ratio') {
+      @include assert-equal(size.size('small'), 0.5rem);
+    }
+
+    @include it('Get unit-adjusted size with string $do') {
+      @include assert-equal(size.size(24px, 'rem'), 1.2rem);
+    }
+
+    @include it('Get unit-adjusted size with list $do') {
+      @include assert-equal(size.size(24px, 'em' 10px), 2.4em);
+    }
+
+    @include it('Run adjustments on the fly with map $do') {
+      @include assert-equal(
+        size.size(
+          24px,
+          (
+            'to': 'em' 10px,
+            'times': 2,
+          )
+        ),
+        4.8em
+      );
+    }
+
+    @include it('Adjusted size with multiple arguments') {
+      @include assert-equal(size.size('medium-page'), 75em);
+    }
+
+    @include it('Get rounded size') {
+      @include assert-equal(size.size('round'), 3rem);
+    }
+
+    @include it('Dont adjust non-comparable units') {
+      @include assert-equal(size.size(24vmin), 24vmin);
+    }
+
+    @include it('Allow viewport units without warning') {
+      @include assert-equal(size.size('viewport'), 10vw);
+    }
+
+    @include it('Allow viewport units without warning') {
+      @include assert-equal(size.size('percent'), 20%);
+    }
+
+    @include it('Allow simple calc values') {
+      @include assert-equal(size.size('calc-simple'), calc(1em + 2vw));
+    }
+
+/*     @include it('Allow compiled calc values') {
```

Comment:
There seems to be something in the test-data map that is being converted incorrectly? I get this error on the last three commented out tests in `size`

```
Error: ("times": 3) isn't a valid CSS value.
   ╷
40 │   @return if(($type == 'string'), $value, '#{$value}');
   │                                              ^^^^^^
   ╵
  sass/utils/_string.scss 40:46                          force()
  sass/utils/_string.scss 115:11                         replace()
  sass/utils/_string.scss 175:16                         interpolate()
  sass/tokens/_parse.scss 185:13                         do()
  sass/tokens/_parse.scss 334:15                         compile-token()
  sass/tokens/_api.scss 295:16                           get()
  sass/scale/_size.scss 55:14                            size()
```
They all appear to be using this section of the test-data map:

  ```
'calc': (
    '%value': 'calc(%s + (math.div(%s, %s)))',
    '%s': ('#root' ('times': 3), '#test-value' ('convert-units': 'vw'), 3),
  ),

Is there a typo in the `calc` map here that's causing these tests to fail @mirisuzanne ?
```